builder/struct.go

```diff
@@ -26,8 +26,37 @@ func (*Struct) Build(gen Generator, ctx *MethodContext, sourceID *xtype.JenID, s
 
```

Comment:
> The constructor can be used to set some defaults, but based on the actual generation if that field (defaulted) is not ignored.. it is going to be overridden even if nil, maybe we could thinkg about an additional setting not-override (or whatever) to skip a field assignment if nil in the source, wdyt? Obv I think this could be a followup issue if you think makes sense

Do you mean if it's a struct pointer then the default value won't be used when the source value is nil?

I'd want to support this use-case:
```go
// goverter:converter
type Converter interface {
    // goverter:init NewOutput
    Convert(source *Input) *Output
}
type Input struct {
    ID int
    FirstName string
    LastName string
}
type Output struct {
    ID int
    FirstName string
    LastName string
}
func NewOutput() *Output {
    return &Output{FirstName: "test"}
}
```
to generate
```go
import example "goverter/example"

type ConverterImpl struct{}

func (c *ConverterImpl) Convert(source *example.Input) *example.Output {
	var pExampleOutput *example.Output = example.NewOutput()
	if source != nil {
		var exampleOutput example.Output
		exampleOutput.ID = (*source).ID
		exampleOutput.FirstName = (*source).FirstName
		exampleOutput.LastName = (*source).LastName
		pExampleOutput = &exampleOutput
	}
	return pExampleOutput
}

```

Reply:
I meant more this use case (taken from test `scenario/constructor_with_pointers.yml`):

```go
        func (c *ConverterImpl) Convert(source *external1.Input) (*external2.Output, error) {
        	var pExternal2Output *external2.Output
        	if source != nil {
        		var external2Output external2.Output = *external2.NewOutputWithDefaults()
        		external2Output.Name = (*source).Name
        		external2Output.ExternalId = (*source).ExternalId
        		var pString *string
        		if (*source).State != nil {
        			xstring := *(*source).State
        			pString = &xstring
        		}
        		external2Output.State = pString
        		pExternal2Output = &external2Output
        	}
        	return pExternal2Output, nil
        }
```

Here consider the case where `NewOutputWithDefaults` set a default value to the `.State` field, as you can see this is field is going to be overridden anyway, even if it is `nil` in the `source`.

This would be fixed if we can move `external2Output.State = pString` inside the `if` stmt, but TBH I do not know which other implication could arise with this.

Reply:
btw I agree that also your use case (when the source is `nil`) should be supported, and it is not right now. Good catch!

Reply:
Yeah, your use-case is reasonable, but I think supporting this isn't that easy because resolving pointer conversions is done separately and doesn't know if we are inside a struct conversion. We could probably add extra code generation to struct.go for pointers to solve this use-case.

This should be done in a separate ticket.

Reply:
https://github.com/jmattheis/goverter/issues/97

Reply:
> This should be done in a separate ticket.

+1 I definitely agree